### ISSUE TITLE
sapper build fails with `TypeError: Cannot read property 'crypto' of undefined` in 0.28.7

### ISSUE THREAD

**Issue by jbg** [sveltejs/sapper#1520]
**Describe the bug**
`sapper build --legacy` fails with `TypeError: Cannot read property 'crypto' of undefined` in Sapper 0.28.7 using Rollup. The same project builds fine with Sapper 0.28.6.

**Logs**
```
[!] (plugin sapper-internal) TypeError: Cannot read property 'crypto' of undefined
TypeError: Cannot read property 'crypto' of undefined
    at .../node_modules/sapper/dist/create_manifest_data.js:3992:95
    at Array.find (<anonymous>)
    at .../node_modules/sapper/dist/create_manifest_data.js:3992:74
    at String.replace (<anonymous>)
    at Object.<anonymous> (.../node_modules/sapper/dist/create_manifest_data.js:3990:57)
    at Generator.next (<anonymous>)
    at .../node_modules/sapper/dist/_commonjsHelpers.js:30:71
    at new Promise (<anonymous>)
    at Object.__awaiter (.../node_modules/sapper/dist/_commonjsHelpers.js:26:12)
    at Object.generateBundle (.../node_modules/sapper/dist/create_manifest_data.js:3962:45)
```

**To Reproduce**
I am working on a small repro and will update this section.

**Information about your Sapper Installation:**
Node 14, yarn & rollup on Linux.

**Severity**
Blocking usage of Sapper entirely on 0.28.7, but downgrading to 0.28.6 is an effective workaround.

**Comment by Conduitry** [comment]
That corresponds to here: https://github.com/sveltejs/sapper/blob/12eb50672b150ae8aae8fa31295515dcef671ef2/src/core/create_compilers/RollupCompiler.ts#L178

Presumably the repro is going to involve doing a dynamic `import('crypto')`. I'm guessing the new dynamic import / CSS handling code doesn't correctly handle dynamic imports of things that don't end up getting bundled. I'm not sure what `c.modules` being undefined indicates.

**Comment by jbg** [comment]
Yes, the project on which this is happening does have a dynamic `import("crypto")`, something along the lines of:

```
var randomBytes;
if (process.browser) {
  var crypto = window.crypto || window.msCrypto;
  if (crypto) {
    randomBytes = new Uint8Array(16);
    crypto.getRandomValues(randomBytes);
  }
}
else {
  const crypto = await import("crypto");
  if (crypto) {
    randomBytes = crypto.randomBytes(16);
  }
}
```

I'll try to use the above as a basis for a minimal repro project later today.

**Comment by benmccann** [comment]
@jbg you could try debugging by changing the line below in `RollupCompiler`:

`const target = <OutputChunk>Object.values(bundle).find(c => !!(<OutputChunk>c).modules[id]);`

To the following:

`const target = <OutputChunk>Object.values(bundle).find(c => (<OutputChunk>c).modules && (<OutputChunk>c).modules[id]);`

Let me know if you're able to either do that or provide a reproduction

**Comment by jbg** [comment]
Your proposed fix does work in 0.28.7. Additionally, the issue no longer occurs in 0.28.8.
